sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnReflector.java

```diff
@@ -241,16 +266,17 @@ private static String format(Method m) {
       final TypeToken<?> in, final TypeToken<?> out) {
     return FluentIterable
         .from(extraProcessContexts.values())
+        .filter(new Predicate<AdditionalParameter>() {
```

Comment:
May be easier to do ".remove(InputProvider)" but this is also more future proof. Maybe "isHidden" (to support future parameters not showing up in docs)?